junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/CsvFileArgumentsProvider.java

```diff
@@ -124,11 +127,14 @@ private CsvParser createCsvParser(InputStream inputStream) {
 
 		private final CsvFileSource annotation;
 
+		private final DefaultConversionProcessor conversionProcessor;
+
 		private Object[] nextCsvRecord;
 
 		CsvParserIterator(CsvParser csvParser, CsvFileSource annotation) {
 			this.csvParser = csvParser;
 			this.annotation = annotation;
+			this.conversionProcessor = getConversionProcessor();
```

Comment:
If this.annotation.nullSymbols() is equals to default value, the conversionProcessor could be skiped.

Reply:
Hi @Ciruman,
thanks for the suggestion! 
Before implementing it I would like to explain why I wrote the code the way I did :)

I tried to implement my solution wit Null-Objects in mind. If the `nullSymbols` is an empty Array I still instantiate a `DefaultConversionProcessor` but because I don't add a `Conversion` to it it won't have any behaviour.
This way I don't have to introduce any more Null-Checks. I think it's a trade-off we have to find.
**Better readability** (imho) **vs Performance** (without making any Benchmarks)

Another way could be to use an `Optional<DefaultConversionProcessor>`.
This would result in a code like this when the conversion needs to be done:

 ```
AtomicReference<String[]> wrapper = new AtomicReference<>(parsedLine);
Object[] processedLine = optionalProcessor.map(defaultConversionProcessor -> defaultConversionProcessor.applyConversions(wrapper.get(), null)).orElse(wrapper.get());
parsedLine = Arrays.copyOf(processedLine, parsedLine.length, String[].class);
```

Which is imho even harder to read.

What are your thoughts?
(cc @sormuras)

Reply:
I follow your line of reasoning and am fine with the better readability your approach provides. If a real performance issue arises from this implementation issue, we may tackle (proof and improve) it later.